src/shared/seccomp-util.c

```diff
@@ -2483,11 +2484,21 @@ int seccomp_suppress_sync(void) {
                                 continue;
                         }
 
-                        r = seccomp_rule_add_exact(
-                                        seccomp,
-                                        SCMP_ACT_ERRNO(0), /* success → we want this to be a NOP after all */
-                                        id,
-                                        0);
+                        if (STR_IN_SET(c, "fdatasync", "fsync", "sync_file_range", "sync_file_range2", "syncfs"))
```

Comment:
Please also update the comment in the beginning of the function: https://github.com/systemd/systemd/pull/34492/commits/73d390be3cb2360d7963aafde684d45cf67d7a86#diff-60141857536525beb87708eaf458438b24ce034aed48172f0db7ee1cd1acc648R2476